design/cdk-bootstrap.md

```diff
@@ -0,0 +1,199 @@
+# cdk-bootstrap
+
+`cdk-bootstrap` is a tool in the AWS CDK toolchain responsible for populating a given environment
+(that is, a combination of AWS account and region)
+with resources required by the CDK to perform deployments into that environment.
+It's a successor to the `cdk bootstrap` sub-command that is part of the `@aws-cdk/core` package,
+and is needed because of the increased bootstrapping requirements present in the
+["Continuous delivery for CDK apps"](https://github.com/aws/aws-cdk/pull/3437) design document.
+
+## Command-line interface
+
+Invoking the `bootstrap` tool looks something like this:
+
+```shell
+$ cdk-bootstrap [--trust accountId[,otherAccountId]...] aws:://account/region
+```
+
+If the `trust` option has been passed on the CLI,
+the tool will present the user with the following "menu":
+
+```shell
+This will create a new IAM Role, trusting the CloudFormation service principal,
+that will be used for deploying your CloudFormation stacks in this environment.
+Please choose the Managed Policy or Policies to attach to the Role
+that contain the permissions required to perform those deployments:
+
+[1]  AdministratorAccess
+[2]  AmazonDynamoDBFullAccess
+[3]  AmazonEC2FullAccess
+[4]  AmazonECS_FullAccess
+[5]  AmazonS3FullAccess
+[6]  AmazonSageMakerFullAccess
+[7]  AmazonSESFullAccess
+[8]  AmazonSNSFullAccess
+[9]  AmazonSQSFullAccess
+[10] AWSLambdaFullAccess
+[11] SimpleWorkflowFullAccess
+...
+
+Please input the number of the chosen Policy from the above list,
+or provide a fully qualified ARN of a different Managed Policy.
+You can input multiple numbers and/or ARNs (up to 10),
+separated by commas.
+
+>
+```
+
+After the user has made their choice,
+or if the `trust` option has not been passed,
+bootstrapping will proceed.
+
+### Bootstrap resources
+
+The `bootstrap` command creates a CloudFormation stack in the environment passed on the command line.
+The resources in that stack are:
+
+* An S3 bucket that holds the file assets and the resulting CloudFormation template to deploy.
```

Comment:
KMS key?

Reply:
Included as a parameter.

Reply:
Just another resource in the template.

Reply:
I wonder, why a key? Our current bootstrapping stores the files unencrypted. Since they're assets, they are taken directly from the source code project, or are built from it. I don't see much value in encrypting them.

Is there a particular reason you want to add a KMS key now?

Reply:
There are many companies that require every bucket to be encrypted, no matter what the contain.

Reply:
Yap, that was my thought.